### PR TITLE
adding db testing blog post

### PR DESCRIPTION
**Pull request by ghost**
db-testing

### CODE REVIEW

**Comment by dhatanian** [comment]
Can you replace the thumbnail with a 335x225px one? This one looks weird when looking at the list of blog posts.

image

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+name: database-testing-with-tsql 
+title: Database Testing With tSQLt
+date: 2016-10-06 10:20:00 +00:00
+author: Halima Koundi
+image:
+   src: /assets/img/custom/blog/db-testing/db-testing.jpg
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:8]: Is there any license/attribution for this image?

Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:8]: Can you replace the thumbnail with a 335x225px one? This one looks weird when looking at the list of blog posts.

image

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+layout: post
+name: database-testing-with-tsql 
+title: Database Testing With tSQLt
+date: 2016-10-06 10:20:00 +00:00
+author: Halima Koundi
+image:
+   src: /assets/img/custom/blog/db-testing/db-testing.jpg
+tags:
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:9]: Can you have a look at existing tags and see if there are some that would match your article?

I've listed the tags and article counts below (there's a script, I'll add it to the site).

If you want to see all the articles for a tag go to

```
https://codurance.com/publications/tag/THE_TAG/
```

```
ira,1
virtual machine,1
Agile,1
oss,1
rest,1
kafka,1
Web API,1
concurrency,1
build system,1
akka-http,1
coordination,1
cli,1
async,1
personal,1
Ubiquitous-language,1
AWS,1
user stories,1
terminal,1
reactive,1
opensource,1
saxon,1
publish-subscribe,1
cloud-services,1
Continuous-Deployment,1
server side,1
data modeling,1
Cucumber,1
csharp,1
test-doubles,1
F#,1
synchronization,1
cqrs,1
Separation of concerns,1
dreams,1
continuous-integration,1
open-source,1
xquery,1
screencast,1
Kata,1
security,1
Babun,1
functional programming,1
kernel,1
Continuous Deployment,1
Java,1
Scala,1
jade,1
sync,1
travis,1
Heroku,1
socrates-uk,1
Walking-Skeleton,1
expresiveness,1
event,1
Domain-Driven-Design,1
virtualisation,1
container,1
interaction-design,1
company-culture,1
Memento,1
Swift,1
ddd,1
TeamCity,1
sbt,1
workflow,1
amazon-web-services,1
javascript,1
scalability,1
efficiency,1
Team,1
open source,1
Security,1
standard,1
swift,1
DDD,1
FsCheck,1
hypervisor,1
correctness,1
incremental-design,1
Terraform,1
open-book management,1
Design,1
product owner,1
messaging,1
organisation,1
decision-making,1
lambda,1
codereview,1
managers,1
google-plus,1
vim,1
rpc,1
haskell,1
cross-platform,1
Kitura,1
tools,1
unit,1
scrum master,1
customisation,1
abstractions,1
C#,1
continuous delivery,1
UI design,1
teamwork,1
inflection-point,1
swagger,1
SOLID,1
Big-Data,1
retrospective,1
c#,1
failure,1
VIPER,1
oauth,1
the-software-craftsman,1
java8,1
apprentice,1
single responsibility,1
operating system,1
ux,1
SBT,1
Web service,1
iteration,1
windows,1
mouseless,1
ui,1
transparency,1
zsh,1
guide,1
NCrunch,1
open financials,1
prompt,1
streams,1
shortcuts,1
.NET,1
ms sql server,1
Mutual Authentication,1
supervision,1
Jenkins,1
outside-in,1
Bounded-context,1
sprint,1
cassandra,1
Travis,1
Amazon-Web-Services,1
deliberate-practice,1
Consolas,1
website-solution,1
event-sourcing,1
bare metal,1
actors,1
estimatation,1
howto,1
collaboration,1
philosophy,1
metrics,1
database,1
unit-test,1
HTTPS,1
Vapor,1
craftsmen,1
Continuous-Integration,1
classicist,1
framework,1
Resharper,1
fsm,1
cohesion,1
socrates,1
command,1
volumes,1
outdoor-retrospective,1
docker,1
Clean architecture,1
big data,2
iOS,2
domain-driven-design,2
Objective-C,2
.net,2
teams,2
threading,2
Learning,2
pair-programming,2
Visual-Studio,2
spring,2
naming,2
integration,2
state,2
streaming,2
maven,2
mentoring,2
anti-pattern,2
spark,2
git,2
scalatra,2
Docker,2
object-orientation-abusers,2
code-smells,2
mob-programming,2
performance,2
android,2
visual-studio,2
REST,2
coupling,2
inflection point,2
test-driven-development,2
change-preventers,2
OOP,2
mentorship,2
Cloud,2
distributed systems,3
refactoring,3
umbraco,3
cms,3
scala,3
cloud,3
career,3
user-story,3
clojure,3
community,3
clean-code,3
devops,3
culture,3
windows-azure,3
productivity,3
serverless,3
aws,3
automation,3
apprenticeship,3
github,3
team,4
software design,4
codurance-way,4
backlog,4
learning,4
practices,4
microservices,5
design-patterns,5
testing,5
scrum,5
TDD,5
akka,5
quality,6
functional-programming,6
architecture,7
codurance,8
agile,11
java,12
tdd,14
design,14
craftsmanship,37
```

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Open the Example.sql file that you will find within the unzipped folder, and execute it in your database server. This will create a test database named `tSQLt_Example`.
+
+Once your database is created, open a new query window on that DB and run the following command:
+`EXEC tSQLt.RunAll`
+This is a command to run all the tests. You should see a failing test in the result screen:
+
+{% img /assets/img/custom/blog/db-testing/db-testing-3.png "" %}
+
+Test case number 11, named test ready for experimentation if 2 particles, is failing. Let’s open the file and see what’s wrong with it; we will need to navigate to the stored procedures folder to find the test.
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:60]: Add `"` or  ``` quotes to the test name to separate it from the rest of the sentence.

Comment (dhatanian) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:60]: Same with the `stored procedures` folder

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+{% img /assets/img/custom/blog/db-testing/db-testing-4.png "" %}
+
+{% img /assets/img/custom/blog/db-testing/db-testing-5.png "" %}
+
+The failing test is calling `IsExperimentReady` and checks that it returns 1 when the particle table has two particles.
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:66]: is called

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Looking at the function, it seems that it is not counting the number of rows in the Particle table, and it is not doing the right checks.
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:67]: Same with `Particle` : use backticks

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+{% img /assets/img/custom/blog/db-testing/db-testing-6.png "" %}
+
+Once we’ve solved the issue, we can run the tests again and see that they are all passing now.
+
+###What is this test doing and how is the fake table working?
+
+Tests in the tSQLt framework are run within transactions. The function we are testing relies on the Particle table to determine whether the system is ready for experimentation. Looking at our test here, we can see that we are creating a fake table of the `Accelerator.Particle` in which we will insert two rows.
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:75]: Use quotes for `Particle` table

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+If we have closer at look at the `tSQLt.FakeTable` SP, it is renaming the original table we are faking, and copying its structure into a new table of the original name without applying any of the constraints.
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:77]: Lazy lazy! use "stored procedure" instead of "SP", more readable

Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:77]: If we have closer **look** at the `tSQLt.FakeTable` stored procedure, **we can see** it is renaming ...

File: _posts/2016-10-06-database-testing-with-tsqlt.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+If we have closer at look at the `tSQLt.FakeTable` SP, it is renaming the original table we are faking, and copying its structure into a new table of the original name without applying any of the constraints. 
+Then we run the test by calling the function under test, knowing that it should use the table we just faked.
+After the test has run wrapped inside the transaction, this transaction is rolled back. This will revert all the changes made to the mocked table.
+The test results that were stored in temporary fields are then saved in a test result table.
+
+{% img /assets/img/custom/blog/db-testing/TestingWithFakeTables.png "" %}
+
+In this first blog post we learned about database testing, and how the tSQLt testing framework works. In the next post, we will go beyond the quick start example and see how to approach legacy transact-SQL, implementing tests into an existing database.
+
```
Comment (dhatanian) [inline _posts/2016-10-06-database-testing-with-tsqlt.md:85]: Not important, but are those three newlines at the end of the file needed?

**Review by dhatanian — CHANGES_REQUESTED** [review]
Great post! Just a few typos and comments. Most important is to fix the image bit.

File: _posts/2016-10-06-database-testing-with-tsqlt.md (hunk 1 above)
Comment (ghost) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:8]: I took the image myself at the office :)

File: _posts/2016-10-06-database-testing-with-tsqlt.md (hunk 2 above)
Comment (ghost) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:9]: yeah maybe testing

File: _posts/2016-10-06-database-testing-with-tsqlt.md (hunk 4 above)
Comment (ghost) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:66]: it is the name of the function under test, I changed the wording

File: _posts/2016-10-06-database-testing-with-tsqlt.md (hunk 8 above)
Comment (ghost) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:85]: they are not indeed.

File: _posts/2016-10-06-database-testing-with-tsqlt.md (hunk 1 above)
Comment (ghost) [reply _posts/2016-10-06-database-testing-with-tsqlt.md:8]: I need help form Giulia for this one, so I will fix on Monday. As I have broken the aspect ratio of the picture when I tried to resize it. I do not have the right editing tool, Paint is limited :)

**Pull request merged**